app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.find_by name: params[:name]&.downcase
```

Comment:
KHông sai nhưng chị có suggest để code clear hơn

em dùng action filter (cụ thể là before_action) để call method load_user
còn method load_user sẽ handle việc tìm user theo name và dồng thời handle nếu user nil luôn

còn methof create ở đây em chỉ handle authenticate và cũng ko cần toán từ an toàn &. `@user&.authenticat` lun nhá